From 88a168fe272e5e63652f17d1553826f312a3db42 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:59 -0400
Subject: [tools] perf/evsel: Synthesize PERF_SAMPLE_TRANSACTION

Message-id: <1408462094-14194-251-git-send-email-jolsa@redhat.com>
Patchwork-id: 88233
O-Subject: [PATCH RHEL7.1 BZ1131394 250/325] perf evsel: Synthesize PERF_SAMPLE_TRANSACTION
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 42d88910c717ba21089251d0ca559abfef0df22d
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Nov 1 15:51:38 2013 +0200

description
===========
Add missing PERF_SAMPLE_TRANSACTION to perf_event__synthesize_sample()
and perf_event__sample_event_size().

This makes the "sample parsing" test pass.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evsel.c b/tools/perf/util/evsel.c
index b121717..5280820 100644
--- a/tools/perf/util/evsel.c
+++ b/tools/perf/util/evsel.c
@@ -1578,6 +1578,9 @@ size_t perf_event__sample_event_size(const struct perf_sample *sample, u64 type,
  if (type & PERF_SAMPLE_DATA_SRC)
   result += sizeof(u64);
 
+ if (type & PERF_SAMPLE_TRANSACTION)
+  result += sizeof(u64);
+
  return result;
 }
 
@@ -1751,6 +1754,11 @@ int perf_event__synthesize_sample(union perf_event *event, u64 type,
   array++;
  }
 
+ if (type & PERF_SAMPLE_TRANSACTION) {
+  *array = sample->transaction;
+  array++;
+ }
+
  return 0;
 }
 
-- 
1.7.1